pyproject.toml

```
@@ -1,19 +1,117 @@
-[tool.black]
-exclude = "3rdparty/*"
-line-length = 100
+[project]
+name = "xgrammar"
+description = "Efficient, Flexible and Portable Structured Generation"
+authors = [{ name = "MLC Team" }]
+readme = "README.md"
+license = { text = "Apache 2.0" }
+classifiers = [
+    "License :: OSI Approved :: Apache Software License",
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Education",
+    "Intended Audience :: Science/Research",
+]
+keywords = ["machine learning", "inference"]
+requires-python = ">=3.9, <4"
+dependencies = [
+    "pydantic",
+    "sentencepiece",
+    "tiktoken",
+    "torch>=2.5.0",
+    "transformers>=4.38.0",
+]
+dynamic = ["version"]
 
-[tool.isort]
-profile = "black"
-src_paths = ["python"]
+[project.urls]
+Homepage = "https://xgrammar.mlc.ai/"
+GitHub = "https://github.com/mlc-ai/xgrammar"
 
-[tool.mypy]
-strict = true
+[project.optional-dependencies]
+triton = [
+    "triton",
+]
+test = [
+    "pytest",
+    "protobuf",
+    "huggingface-hub[cli]",
+]
+
+[tool.scikit-build.metadata.version]
+provider = "scikit_build_core.metadata.regex"
+input = "python/xgrammar/version.py"
+
+[build-system]
+requires = [
+    "scikit-build-core>=0.10.0",
+    "pybind11>=2.11.0",
+]
+build-backend = "scikit_build_core.build"
+
+[tool.scikit-build]
+minimum-version = "build-system.requires"
+
+# Build configuration
+build-dir = "build"
+build.verbose = true
+
+# CMake configuration
+cmake.version = "CMakeLists.txt"
+cmake.args = []
+cmake.build-type = "RelWithDebInfo" # Release is Default
```

Setting RelWithDebugInfo as default is fine.
```suggestion
cmake.build-type = "RelWithDebInfo"
```